### PR TITLE
[cert] Add fuzz tests for the X.509 parser

### PR DESCRIPTION
**Pull request by mcy**
Pending on #81

### CODE REVIEW

**Review by cfrantz — APPROVED** [review]

File: fuzz/targets/x509_unsigned.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+use manticore::cert;
+use manticore::cert::Algo;
+use manticore::cert::Cert;
+use manticore::cert::CertFormat;
+use manticore::cert::PublicKeyParams;
+use manticore::crypto::sig;
+
+/// A `Ciphers` that blindly accepts all signatures.
+struct NoVerify;
```
Comment (alistair23) [inline fuzz/targets/x509_unsigned.rs:20]: Can you not reused the existing `NoVerify`?

File: fuzz/targets/x509_unsigned.rs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+fuzz_target!(|data: &[u8]| {
+    let _ = Cert::parse(
+        data,
+        CertFormat::RiotX509,
+        None,
+        &mut NoVerify,
+    );
+
+    // NOTE: we might actually succeed at creating a valid cert, so we can't
+    // check for is_err() here.
```
Comment (alistair23) [inline fuzz/targets/x509_unsigned.rs:53]: What am I missing here? Won't this always succeed?

Comment (mcy) [reply fuzz/targets/x509_unsigned.rs:53]: libfuzzer might accidentally explore a path that produces valid certificates, which is not a reason to crash the test.

File: fuzz/targets/x509_unsigned.rs (hunk 1 above)
Comment (mcy) [reply fuzz/targets/x509_unsigned.rs:20]: That would require exposing it as part of Manticore's API. If this starts happening enough I may consider putting it somewhere global, but definitely not in manticore.

**Pull request merged**